src/main/java/core/basesyntax/HelloWorld.java

```diff
@@ -1,8 +1,19 @@
 package core.basesyntax;
 
-/**
- * Feel free to remove this class and create your own.
- */
 public class HelloWorld {
+    private static final int NUMBER_OF_FIGURES = 6;
 
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[NUMBER_OF_FIGURES];
+        for (int i = 0; i < figures.length; i++) {
+            if (i >= figures.length / 2) {
+                figures[i] = figureSupplier.getDefaultFigure();
+            } else {
+                figures[i] = figureSupplier.getRandomFigure();
+            }
+            System.out.println(figures[i].draw());
```

Comment:
Avoid creating new instances of `FigureSupplier` in a loop. Since `figureSupplier` is already created outside the loop, use that instance inside the loop.